ascent/ascent/cli.py

```diff
@@ -0,0 +1,194 @@
+import mimetypes
+import os
+from pathlib import Path
+
+import click
+import girder_client
+
+# TODO
+# Upload annotations
+# -validate file
+# -upload annotation file
+# Update existing folder
+#  - require id
+# -update/overwrite
+# Support tags
+# -option for upload and update
+# Large image
+
+supportedVideoTypes = [
+  'video/mp4',
+  'video/webm',
+  'video/ogg',
+]
+supportedImageTypes = [
+  'image/apng',
+  'image/avif',
+  'image/bmp',
+  'image/gif',
+  'image/jpeg',
+  'image/png',
+  'image/tiff',
+  'image/webp',
+]
+
+_global_options = [
+    click.option(
+        "--port",
+        type=int,
+        help="Provide port that DIVE is running on. Default 8010",
+    ),
+    click.option(
+        "--url",
+        type=str,
+        help="Provide api URL. Defaults to 'localhost'"
+    )
+]
+
+
+def global_options(func):
+    for option in _global_options:
+        func = option(func)
+    return func
+
+
+_upload_options = [
+    click.option(
+        "--path",
+        required=True,
+        type=click.Path(exists=True, readable=True, path_type=Path),
+        help="Local location of items(s) to upload"
+    ),
+    click.option(
+        "--parent-folder",
+        type=str,
+        help="""
+        ID of girder folder that will be the parent of created folder.
+        Defaults to user root.
+        """
+    ),
+    click.option(
+        "--folder-name",
+        type=str,
+        required=True,
+        help="Unique folder name"
+    )
+
+]
+
+
+def upload_options(func):
+    for option in _upload_options:
+        func = option(func)
+    return func
+
+
+apiURL = "localhost"
+
+
+def login(api, port):
+    apiURL = api if api else "localhost"
+    local_port = port if port else 8010
+    gc = girder_client.GirderClient(
+        apiURL,
+        port=local_port,
+        apiRoot="girder/api/v1"
+      )
+    if not os.environ.get('DIVE_USER') or not os.environ.get('DIVE_PW'):
+        gc.authenticate(interactive=True)
+    else:
+        gc.authenticate(os.environ.get('DIVE_USER'), os.environ.get('DIVE_PW'))
+    return gc
+
+
+@click.group()
+def ascent():
+    pass
+
+
+def create_folder(parent_folder, folder_name, gc):
+    new_folder = None
+    try:
+        if parent_folder:
+            new_folder = gc.createFolder(
+                parent_folder,
+                folder_name,
+                parentType="folder"
+              )
+        else:
+            me = gc.get("user/me")
```

Comment:
Just an observation.  Most users on VIAME stick to the public/private structure for their data.  Should this instead default to private and have an easy optional flag/setting which can override the default private.  Meaning that before uploading a bunch of stuff a person could change the default upload location to another area using an environmental setting?  Just and idea and doesn't need to be implemented.

Reply:
That's a good point and I might change it in the future but I think I will keep it as is for now. My thought was having the default be visible in the user root would be easy to locate especially since it is set up for a local spin-up